### ISSUE TITLE
Segfault when token creation fails during auth

### ISSUE THREAD

**Issue by eyakubovich** [gophercloud/gophercloud#109]
When creating a token via auth endpoint, the call can fail (e.g. due to network problems) but the error is not checked, causing a nil pointer to be dereferenced.

**Comment by jrperritt** [comment]
Do you have the trace from the segfault? The error should be getting checked in `gophercloud.ExtractInto` before any analysis is done on `result`.

**Comment by eyakubovich** [comment]
The real problem is that `openstack/identity/v3/tokens/requests.go:Create` will not set `r.Err` if `c.Post` fails. Therefore `ExtractInfo` will not see the error. Here's the trace:

```
goroutine 68 [running]:
panic(0x7c2400, 0xc82000a130)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
testing.tRunner.func1(0xc820188090)
	/usr/local/go/src/testing/testing.go:467 +0x192
panic(0x7c2400, 0xc82000a130)
	/usr/local/go/src/runtime/panic.go:443 +0x4e9
github.com/gophercloud/gophercloud/openstack/identity/v3/tokens.commonResult.ExtractToken(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/eyakubovich/devel/go/src/github.com/gophercloud/gophercloud/openstack/identity/v3/tokens/results.go:63 +0x154
github.com/gophercloud/gophercloud/openstack.v3auth(0xc820188120, 0xc8201668a0, 0x22, 0x7faf4fa2a3c8, 0xc82019a280, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/eyakubovich/devel/go/src/github.com/gophercloud/gophercloud/openstack/client.go:157 +0x192
github.com/gophercloud/gophercloud/openstack.Authenticate(0xc820188120, 0x8afe80, 0x21, 0x841a00, 0x2, 0x0, 0x0, 0x842600, 0x6, 0x0, ...)
	/home/eyakubovich/devel/go/src/github.com/gophercloud/gophercloud/openstack/client.go:81 +0x481
github.com/gophercloud/gophercloud/openstack.AuthenticatedClient(0x8afe80, 0x21, 0x841a00, 0x2, 0x0, 0x0, 0x842600, 0x6, 0x0, 0x0, ...)
	/home/eyakubovich/devel/go/src/github.com/gophercloud/gophercloud/openstack/client.go:58 +0xbc
github.com/gophercloud/gophercloud/openstack/testing.testAuthenticatedClientFails(0xc820188090, 0x8afe80, 0x21)
	/home/eyakubovich/devel/go/src/github.com/gophercloud/gophercloud/openstack/testing/client_test.go:303 +0x6f
github.com/gophercloud/gophercloud/openstack/testing.TestAuthenticatedClientV3Fails(0xc820188090)
	/home/eyakubovich/devel/go/src/github.com/gophercloud/gophercloud/openstack/testing/client_test.go:310 +0x36
```

**Comment by eyakubovich** [comment]
I ended up changing the PR to set the error after `Post` and letting `ExtractInto` handle the error

**Comment by jrperritt** [comment]
closed by #92
